packages/augur-sdk/src/utils.ts

```diff
@@ -112,62 +111,97 @@ export function logError(
   if (err != null) {
     console.error(err);
     if (result != null) console.log(result);
+
   }
 }
 
+export interface PayoutNumeratorValue {
+  malformed?: boolean;
+  invalid?: boolean;
+  outcome?: string;
+}
+
 export function calculatePayoutNumeratorsValue(
   displayMaxPrice: string,
   displayMinPrice: string,
   numTicks: string,
   marketType: string,
   payout: string[]
-): string | null {
-  const isScalar = marketType === MarketTypeName.Scalar;
-
-  if (!payout) return null;
-  if (payout.length === 0) return null;
-
-
-  if (isScalar) {
+): PayoutNumeratorValue {
+  if (marketType === MarketTypeName.Scalar) {
     if (!isWellFormedScalar(payout)) {
-      return MALFORMED_OUTCOME;
+      return { malformed: true };
+    }
+
+    if (Number(payout[0]) > 0) {
+      return { invalid: true };
     }
 
     const longPayout = new BigNumber(payout[1]);
     const priceRange = new BigNumber(displayMaxPrice, 10).minus(new BigNumber(displayMinPrice, 10));
     // calculation: ((longPayout * priceRange) / numTicks) + minPrice
-    return longPayout
+    const outcome = longPayout
       .times(priceRange)
       .dividedBy(new BigNumber(numTicks, 10))
       .plus(new BigNumber(displayMinPrice, 10))
       .toString();
+    return { outcome };
   } else {
-    if (!isWellFormedCategorical(payout)) { // or yes/no
-      return MALFORMED_OUTCOME;
+    switch(marketType) {
+      case MarketTypeName.Categorical:
+        if (!isWellFormedCategorical(payout)) return { malformed: true };
+        break;
+      case MarketTypeName.YesNo:
+        if (!isWellFormedYesNo(payout)) return { malformed: true };
+        break;
+      default: return { malformed: true }; // bad market type
     }
 
-    return payout.findIndex((item: string) => Number(item) > 0).toString();
+    const outcome = payout.findIndex((item: string) => Number(item) > 0);
+    if (outcome === 0) {
+      return { invalid: true };
+    } else {
+      return { outcome: String(outcome) };
+    }
   }
 }
 
-function isWellFormedCategorical(payout: string[]): boolean {
-  // A categorical or Yes/No payout is well-formed if:
-  // 1. Exactly one of its payouts is non-zero.
+export function isWellFormedYesNo(payout: string[]): boolean {
+  // A Yes/No payout is well-formed if:
+  // 1. There are exactly 3 payout values.
+  // 2. Exactly one of its payouts is non-zero.
+
+  if (payout.length !== 3) return false;
+  return countNonZeroes(payout) === 1;
+}
+
+export function isWellFormedCategorical(payout: string[]): boolean {
+  // A categorical is well-formed if:
+  // 1. There are between 3 and 9 payout values (2-8 plus invalid)
```

Comment:
can only be 8 items in a payout array for categorical
0 // invalid
and 7 possible others `not 9`

should say (2-7 plus invalid)

Reply:
oh!